lib/superset/dashboard/bulk_delete_cascade.rb

```diff
@@ -31,7 +31,7 @@ def perform
       private
 
       def delete_datasets(dashboard_id)
-        datasets_to_delete = Superset::Dashboard::Datasets::List.new(dashboard_id).datasets_details.map{|d| d[:id] }
+        datasets_to_delete = Superset::Dashboard::Datasets::List.new(dashboard_id: dashboard_id).datasets_details.map{|d| d[:id] }
         Superset::Dataset::BulkDelete.new(dataset_ids: datasets_to_delete).perform if datasets_to_delete.any?
```

Comment:
@soundarya-mv 

flagging that we will need another Ticket to not allow a BulkDelete 
if a chart or dataset is used on a different dashboard.

as currently there are no checks around this